renderer/containers/UI/FormattedDateTime.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { FormattedDate, FormattedTime } from 'react-intl'
+import { settingsSelectors } from 'reducers/settings'
+
+const mapStateToProps = state => ({
+  clockFormat24: settingsSelectors.currentConfig(state).clockFormat24,
+})
+
+const FormattedDateTime = ({ clockFormat24, format, month = 'long', value }) => {
+  switch (format) {
+    case 'date':
+      return <FormattedDate day="2-digit" month={month} value={value} year="numeric" />
+    case 'time':
+      return <FormattedTime hour12={!clockFormat24} value={value} />
+    default:
+      return (
+        <FormattedDate
+          day="2-digit"
+          hour="numeric"
+          hour12={!clockFormat24}
+          minute="numeric"
+          month={month}
+          value={value}
+          year="numeric"
+        />
+      )
+  }
+}
+
+FormattedDateTime.propTypes = {
+  clockFormat24: PropTypes.string,
```

Comment:
This should be `PropTypes.bool` in current setup, though as noted earlier Id suggest changing the setting from a bool to an enum string.